### PR TITLE
Change password from panel

### PR DESCRIPTION
**Pull request by SebastiaanKloos**
CleanShot 2021-04-28 at 14 40 37

### CODE REVIEW

**Review by Glennmen — APPROVED** [review]

**Comment by Cannonb4ll** [comment]
Works really well, however, wouldn't it cool if we extend this a tad more with the new `Password` Rule that has just been added to the latest Laravel version?

I recommend setting the validation rule too:

```php
            'password' => [
                'required',
                'confirmed',
                Password::min(8)
                    ->letters()
                    ->numbers()
                    ->uncompromised()
            ]
```

**Comment by Glennmen** [comment]
@Cannonb4ll Cool idea! Maybe also set a threshold for `uncompromised()` 🤔 treat it as compromised only if it was found +3 times? To filter out commonly used passwords.

**Comment by SebastiaanKloos** [comment]
@Cannonb4ll @Glennmen I've added the rules as suggested. Also, the registration form uses the new validation rules!

File: app/Http/Controllers/Profile/ProfileController.php [hunk 1]
```diff
@@ -38,6 +41,39 @@ public function update(Request $request)
...
+                    ->numbers()
+                    ->uncompromised()
+            ],
+        ]);
+
+        if ($validator->fails()) {
+            return redirect()->route('panel.profile.show')->with('error', $validator->errors()->first());
+        }
+
+        if (! Hash::check($request->current, $request->user()->password)) {
```
Comment (Glennmen) [inline app/Http/Controllers/Profile/ProfileController.php:66]: Is this actually needed? Can't it be done with validation rules? https://laravel.com/docs/8.x/validation#rule-password

**Review by Glennmen — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/Profile/ProfileController.php (hunk 1 above)
Comment (SebastiaanKloos) [reply app/Http/Controllers/Profile/ProfileController.php:66]: @Glennmen don't really know... I'll try it soon

**Pull request merged**